### ISSUE TITLE
Fixed positioning does not work

### ISSUE THREAD

**Issue by RobertsNeil24** [jakiestfu/Snap.js#79]
Hi jakiestfu,

Firstly, great plugin.
However, I am trying to fix the toggle button to the top of the page so it behaves like a sticky header and follows the user as they scroll down the page. 

When I add position:fixed to the element it does not position at the top of the page, but even worse is that the anchor tag becomes unclickable. I thought I could solve this with z-index but no luck. I think it might have something to do with overflow:auto on the content. 
Please help - as this is a deal breaker. 

Thanks.

**Comment by TommyKolkman** [comment]
I've got this problem too, it has something to do with the overflow:auto. The overflow:auto kills all jQuery scroll functionality as well. I made a fixed navigation bar that followed your scroll position, basically, and now that's broken. Any ideas?

**Comment by jakiestfu** [comment]
Would you guys mind providing a quick demo?

**Comment by RobertsNeil24** [comment]
Hi Jacob,

I have put together a very rough demo.
http://codepen.io/anon/pen/wFiGg

I can't seem to link to your snap.js file, but the code is all there so you can get the general idea.

Thanks.

**Comment by TommyKolkman** [comment]
I've got a demo setup at http://elephant.elephantlabs.nl/snap_test/over_ons.html

Snap.js is working here, as you can see, and the included script jquery.theme.js provides the scroll logic. The bar #SubMenu should stay fixed when the browser reaches its offset.

Is this demo sufficient for you?

**Comment by jakiestfu** [comment]
@DatoCape @tommykolkman Is this not a matter of creating the proper layout than Snap.js being broken?

Normally, in a web app, even a responsive app, I have the following:

```
#header-toolbar.static
#content.scrollable
#footer-toolbar.static
```

VS the layout you're proposing (if I'm not mistaken)

```
#content.scrollable
    #header-toolbar.fixed
    #footer-toolbar.fixed
```

The proposed layouts for Snap.js adhere to the top set of styles. Personally, i think it should be done that way so you don't need any fixed elements which mobile devices can have a hard time with...

Thoughts?

**Comment by RobertsNeil24** [comment]
On the one hand I agree that fixed elements are not the best for mobile devices, however on the other I think that a fixed header on a desktop layout is great for good UI and UX design principles as the navigation is always displayed for the user.

I have even tried to implement snap.js for only mobile layouts by detecting screen resolution... however by using this plugin, it prevents the desktop layout from implementing a fixed header as well? Perhaps you may have a better solution?

Thanks.

**Comment by TommyKolkman** [comment]
@jakiestfu You are right that the static header should probably be outside of the #content.scrollable. However, at some point, it should stick, something I detect via using $(window).scroll();. 

This, however, is not working anymore. I do understand the scope has changed because I'm scrolling inside of #content.scrollable now. $('#content.scrollable').scroll(); is not firing though, that's my big problem... Any thoughts on that?

**Comment by timmeade** [comment]
was this every addressed?   Hitting with bootstrap 3 navbar-fixed-top.   Seems to be same issue.

**Comment by j5uh** [comment]
@TimMeade I'm with you in the same boat. I'm using bootstrap 3 and ever fixed element is not working. :/

**Comment by ericpeters0n** [comment]
Ditto, bootstrap navbars break. Suggested css workarounds to achieve desired behavior (i.e. fixed navbar and footers in snap-content)?

While not consistent with the canonical Facebook drawer example or most other apps (where the navbars and footers slide as well), @jakiestfu's suggested refactoring of the layout above does work in bootstrap3... It just means your drawers get less real estate.

**Comment by j5uh** [comment]
The workaround I came up with was moving the header/nav out of the absolute containers. You can see what I did here: 

http://www.blinds.com/mobile

**Comment by ebuildy** [comment]
This is a very old chronium bug:

Chrome cannot render position:fixed on elements under a transformation (http://code.google.com/p/chromium/issues/detail?id=20574)

**Comment by dyba** [comment]
:watch:

**Comment by ms88privat** [comment]
you dont need a fixed position nav-bar. position it absolute within the snap content. 
Then add another wrapper in the snap content with:
#wrapper {
    position:absolute; 
    z-index:1;
    top: {{height of nav-bar}} px; left:0;
    width:100%;
    overflow:auto;
}
and apply something like iScroll.js to the wrapper for scrolling the content inside the wrapper ONLY. This way the nav-bar will allways stay visible and will slide away with the content if the drawer opens.

**Comment by OliverKK** [comment]
@ms88privat 
Do u have a example for your thoughts?

**Comment by pcompassion** [comment]
I'm having the same problem...

**Comment by roymj88** [comment]
@jakiestfu : You are correct in your own ways regarding the structure of DOM. But consider a single page application where you need to envelope the entire thing inside a container. Take a list view for example, which is inside snap content. I need to get the scroll event of the list. This will not work as the the DOM no longer recognizes the scroll, position properties any more.

```
$(window).on('scroll', function(){
alert('This doesnt get called.");
});
```

Kindly do address this as many of the users are having the same issue and its currently been over-looked. What you've created is an awesome plugin and it would be a shame if we cannot find a way around this.

Cheers
Roy

**Comment by frankie-loves-jesus** [comment]
Check out jQuery Mobile fixed headers with iScroll:

http://jsfiddle.net/frank_o/jd2du1ga/

**Comment by heyaustinwood** [comment]
Any other solutions besides utilizing @frankie-loves-jesus iScroll option?

**Comment by andreaguarino** [comment]
+1

**Comment by EmilyHerr** [comment]
Position: sticky seems to work on 'fixed' navigation inside the snap area, but it is not supported by any IE browser so it's probably not a viable solution. This really needs to be looked at, though. It looks very odd if the navbar doesn't slide over.
